### PR TITLE
Update to swagger-parser to 2.0.14-OpenAPITools.org-1

### PR DESCRIPTION
**Pull request by jmini**
### PR checklist

- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [ ] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [ ] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Update the Swagger-Parser version to `2.0.14-OpenAPITools.org-1`

Previous PR #3239

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -271,6 +271,7 @@
                                 </requireMavenVersion>
                                 <requireReleaseDeps>
                                     <message>No Snapshots Allowed!</message>
+                                    <onlyWhenRelease>true</onlyWhenRelease>
```
Comment (jmini) [inline pom.xml]: REMOVE THIS LINE when we switch to a released version!

**Comment by jmini** [comment]
The CI jobs looks good to me.

* The samples must be update (when the schema is `integer`, the new version of swagger-parser/swagger-core adds a format -- this is making some internal default value explicit => OK for me)
* Some CI services can not handle `SNAPHSOT` versions

---
I have just created the release:
https://github.com/OpenAPITools/swagger-parser/releases/tag/v2.0.14-OpenAPITools.org-1

It will appear on maven central soon.

---
I will update this PR when the jars are on Maven Central.

File: samples/client/petstore/go/go-petstore-withXml/api/openapi.yaml [hunk 2]
```diff
@@ -791,6 +791,7 @@ paths:
               properties:
                 integer:
                   description: None
+                  format: int32
```
Comment (JFCote) [inline samples/client/petstore/go/go-petstore-withXml/api/openapi.yaml]: I don't think these changes are related to the PR?

Comment (JFCote) [reply samples/client/petstore/go/go-petstore-withXml/api/openapi.yaml]: Oh I see you updated the samples, nevermind then!

Comment (jmini) [reply samples/client/petstore/go/go-petstore-withXml/api/openapi.yaml]: I guess either Swagger-Parser is setting those (explicit default), or Swagger-Core has a new behavior.

IMO this is an acceptable change.

File: samples/server/petstore/rust-server/output/petstore-with-fake-endpoints-models-for-testing/api/openapi.yaml [hunk 3]
```diff
@@ -726,6 +726,7 @@ paths:
               properties:
                 integer:
                   description: None
+                  format: int32
```
Comment (wing328) [inline samples/server/petstore/rust-server/output/petstore-with-fake-endpoints-models-for-testing/api/openapi.yaml]: FYI. Our petstore (fake) spec does not specify the format. I think swagger parser now defaults format to int32 for integer type if no format is specified.

Not sure if the community welcomes this change. We'll see.

**Pull request merged**